pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/utils.py

```diff
@@ -99,3 +100,22 @@ def format_row(row: Mapping[str, Any]) -> str:
         result.append(format_row(row))
 
     return "\n".join(result)
+
+def fix_ipv6(host: str) -> str:
+    """Fix up host for SSH when an IPv6 address is used.
+
+    It returns an SSH-compatible host string if the host is using an
+    IPv6 address, otherwise it returns the passed host string unmodified.
+    """
+    try:
+        host_split = host.split("@")
+        host_fixed = host_split[-1].strip("[]").replace("%25", "%")
+        IPv6Address(host_fixed)
+        if len(host_split) > 1:
+          host = host_split[0] + "@" + host_fixed
+        else:
+          host = host_fixed
+    except AddressValueError:
+        host = host
+
+    return host
```

Comment:
Please add some tests for this function, otherwise I have zero idea what this is meant to do.

Reply:
Thanks, fixed.